### PR TITLE
[5.0] New 'callConsole' shortcut for unit tests

### PR DESCRIPTION
**Pull request by rkgrep**
Syntactic sugar for artisan testing

### CODE REVIEW

File: src/Illuminate/Foundation/Testing/ApplicationTrait.php [hunk 1]
```diff
@@ -163,6 +170,19 @@ public function be(UserContract $user, $driver = null)
 	}
 
 	/**
+	 * Call console command and return code
+	 *
+	 * @param       $command
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/ApplicationTrait.php]: typehint required

File: src/Illuminate/Foundation/Testing/ApplicationTrait.php [hunk 2]
```diff
@@ -20,6 +20,13 @@
 	protected $response;
 
 	/**
+	 * The last code returned by console
+	 *
+	 * @var integer
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/ApplicationTrait.php]: int, not integer

File: src/Illuminate/Foundation/Testing/ApplicationTrait.php [hunk 3]
```diff
@@ -163,6 +170,19 @@ public function be(UserContract $user, $driver = null)
 	}
 
 	/**
+	 * Call console command and return code
+	 *
+	 * @param       $command
+	 * @param array $parameters
+	 *
+	 * @return integer
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/ApplicationTrait.php]: same

**Comment by GrahamCampbell** [comment]
Also needs squash to one commit.

**Comment by taylorotwell** [comment]
Pretty decent idea thought not sure I like the method name.

**Comment by GrahamCampbell** [comment]
Maybe "callCommand"?

**Comment by rkgrep** [comment]
I had that idea, but I think that callCommand should refer to Command Bus functionality.
Another option is execArtisan or callArtisan.

P.S. I've prepaired changes commented above. Waiting for your descision about method name

**Comment by GrahamCampbell** [comment]
> callCommand should refer to Command Bus functionality.

Not really. Your calling a console **command**. Not a "console".

**Comment by rkgrep** [comment]
I agree, but even in generators there are 2 commands: **make:console** and **make:command**.
Actually the method executes only artisan commands, so **callArtisan** is better similarly to Artisan::call() facade (which also has similiar arguments and result).

**Pull request merged**